src/Umbraco.Core/Cache/ContentCacheRefresher.cs

```diff
@@ -160,17 +164,24 @@ internal static void NotifyPublishedSnapshotService(IPublishedSnapshotService se
     public class JsonPayload
     {
         public JsonPayload(int id, Guid? key, TreeChangeTypes changeTypes)
+            : this(id, key, changeTypes, false)
+        { }
+
+        public JsonPayload(int id, Guid? key, TreeChangeTypes changeTypes, bool blueprint = false)
         {
             Id = id;
             Key = key;
             ChangeTypes = changeTypes;
+            Blueprint = blueprint;
```

Comment:
@ronaldbarendse would it be better to use `UmbracoObjectTypes.DocumentBlueprint`

https://github.com/umbraco/Umbraco-CMS/blob/e04c4a89adcbe73d35290a993713a5faf4dc55ff/src/Umbraco.Core/Models/UmbracoObjectTypes.cs#L167

or a `Guid? EntityType` like this?
https://github.com/umbraco/Umbraco-CMS/blob/e04c4a89adcbe73d35290a993713a5faf4dc55ff/src/Umbraco.Core/Models/Notification.cs#L19

in case it need to handle something else than blueprints / content templates in future? Maybe something special for entity containers (folders)?

Reply:
I don't think that's necessary, because the ID/key will only reference `IContent` and using a boolean makes the checks a lot easier/faster as well 😄